packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/remote_constraint_provider.dart

```diff
@@ -33,9 +42,99 @@ class DefaultRemoteLoggingConstraintProvider
   final DefaultRemoteConfiguration _config;
   final AWSCredentialsProvider _credentialsProvider;
 
+  LoggingConstraint? _loggingConstraint;
+  DateTime? _lastUpdated;
```

Comment:
not sure why we need this.

Reply:
The `_loggingConstraint` is used to cache the logging constraint and `_lastUpdated` is used to check to see if the logging constraint is older than the refresh interval.